core/src/main/java/com/radixdlt/consensus/liveness/RotateOnceDecorator.java

```diff
@@ -0,0 +1,131 @@
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.consensus.liveness;
+
+import com.google.common.collect.Ordering;
+import com.google.common.primitives.UnsignedBytes;
+import com.radixdlt.consensus.bft.BFTValidator;
+import com.radixdlt.consensus.bft.BFTValidatorId;
+import com.radixdlt.consensus.bft.BFTValidatorSet;
+import com.radixdlt.consensus.bft.Round;
+import com.radixdlt.rev2.ComponentAddress;
+import com.radixdlt.utils.KeyComparator;
+import java.util.*;
+
+/**
+ * A {@link ProposerElection} decorator which first simply iterates over all validators once (in the
+ * "highest stake first" order), and then lets the underlying implementation take over.
+ */
+public final class RotateOnceDecorator implements ProposerElection {
```

Comment:
This approach is only asymptotically "fair", compared with fixing the first n rounds of WeightedRotatingLeader rotation (and letting the leaders with more stake build up "power" over that time) - or directly using the method of equal proportions.

But given that we're separating proposal count from emissions, this is probably fine.

Reply:
I agree this is probably less fair than MoEP, but it is definitely a more gradual step (i.e. it affects only `validatorsSet.size()` first rounds of an epoch, so even if a massive bug is there, it does not destroy everything), which I value.

Let me merge this now, and we will still have opportunities to tune it later